sdks/go/pkg/beam/runners/dataflow/dataflow.go

```diff
@@ -242,6 +252,12 @@ func getJobOptions(ctx context.Context) (*dataflowlib.JobOptions, error) {
 		}
 	}
 
+	if *flexRSGoal != "" {
+		if *flexRSGoal != "FLEXRS_UNSPECIFIED" && *flexRSGoal != "FLEXRS_SPEED_OPTIMIZED" && *flexRSGoal != "FLEXRS_COST_OPTIMIZED" {
+			return nil, errors.New("invalid flex resource scheduling goal. Use --flexrs_goal=(FLEXRS_UNSPECIFIED|FLEXRS_SPEED_OPTIMIZED|FLEXRS_COST_OPTIMIZED)")
+		}
```

Comment:
Nit: The switch syntax, while it inverts the case, is a bit clearer since it avoids repeating the `*flexRSGoal !=` several times.

Also, print out the flag value we did get, which makes it easier for users to spot typos.

```suggestion
		switch *flexRSGoal {
		case "FLEXRS_UNSPECIFIED" , "FLEXRS_SPEED_OPTIMIZED", "FLEXRS_COST_OPTIMIZED":
		  // valid values
		default:
			return nil, errors.Errorf("invalid flex resource scheduling goal. Got %q; Use --flexrs_goal=(FLEXRS_UNSPECIFIED|FLEXRS_SPEED_OPTIMIZED|FLEXRS_COST_OPTIMIZED)", *flexRSGoal)
		}
```